File: Serene/Serene.Script/Imports/ServiceContracts/Northwind.OrderDetailRow.cs

namespace Serene.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    [Imported, Serializable, PreserveMemberCase]
    public partial class OrderDetailRow
    {
        [InlineConstant] public const string IdProperty = "DetailID";
        [InlineConstant] public const string LocalTextPrefix = "Northwind.OrderDetail";
    
        public Int32? DetailID { get; set; }
        public Int32? OrderID { get; set; }
        public Int32? ProductID { get; set; }
        public Decimal? UnitPrice { get; set; }
        public Int16? Quantity { get; set; }
        public Single? Discount { get; set; }
        public String OrderCustomerID { get; set; }
        public Int32? OrderEmployeeID { get; set; }
        public String OrderDate { get; set; }
        public String OrderShippedDate { get; set; }
        public Int32? OrderShipVia { get; set; }
        public String OrderShipCity { get; set; }
        public String OrderShipCountry { get; set; }
        public String ProductName { get; set; }
        public Boolean? ProductDiscontinued { get; set; }
        public Int32? ProductSupplierID { get; set; }
        public String ProductQuantityPerUnit { get; set; }
        public Decimal? ProductUnitPrice { get; set; }
        public Decimal? LineTotal { get; set; }
    
        [Imported, PreserveMemberCase]
        public static class Fields
        {
            [InlineConstant] public const string DetailID = "DetailID";
            [InlineConstant] public const string OrderID = "OrderID";
            [InlineConstant] public const string ProductID = "ProductID";
            [InlineConstant] public const string UnitPrice = "UnitPrice";
            [InlineConstant] public const string Quantity = "Quantity";
            [InlineConstant] public const string Discount = "Discount";
            [InlineConstant] public const string OrderCustomerID = "OrderCustomerID";
            [InlineConstant] public const string OrderEmployeeID = "OrderEmployeeID";
            [InlineConstant] public const string OrderDate = "OrderDate";
            [InlineConstant] public const string OrderShippedDate = "OrderShippedDate";
            [InlineConstant] public const string OrderShipVia = "OrderShipVia";
            [InlineConstant] public const string OrderShipCity = "OrderShipCity";
            [InlineConstant] public const string OrderShipCountry = "OrderShipCountry";
            [InlineConstant] public const string ProductName = "ProductName";
            [InlineConstant] public const string ProductDiscontinued = "ProductDiscontinued";
            [InlineConstant] public const string ProductSupplierID = "ProductSupplierID";
            [InlineConstant] public const string ProductQuantityPerUnit = "ProductQuantityPerUnit";
            [InlineConstant] public const string ProductUnitPrice = "ProductUnitPrice";
            [InlineConstant] public const string LineTotal = "LineTotal";
        }
    }
    
}

File: Serene/Serene.Web/Modules/Imports/ServerTypings/Northwind.OrderDetailRow.ts
namespace Serene.Northwind {
    export interface OrderDetailRow {
        DetailID?: number;
        OrderID?: number;
        ProductID?: number;
        UnitPrice?: number;
        Quantity?: number;
        Discount?: number;
        OrderCustomerID?: string;
        OrderEmployeeID?: number;
        OrderDate?: string;
        OrderShippedDate?: string;
        OrderShipVia?: number;
        OrderShipCity?: string;
        OrderShipCountry?: string;
        ProductName?: string;
        ProductDiscontinued?: boolean;
        ProductSupplierID?: number;
        ProductQuantityPerUnit?: string;
        ProductUnitPrice?: number;
        LineTotal?: number;
    }

    export namespace OrderDetailRow {
        export const idProperty = 'DetailID';
        export const localTextPrefix = 'Northwind.OrderDetail';

        export namespace Fields {
            export declare const DetailID: 'DetailID';
            export declare const OrderID: 'OrderID';
            export declare const ProductID: 'ProductID';
            export declare const UnitPrice: 'UnitPrice';
            export declare const Quantity: 'Quantity';
            export declare const Discount: 'Discount';
            export declare const OrderCustomerID: 'OrderCustomerID';
            export declare const OrderEmployeeID: 'OrderEmployeeID';
            export declare const OrderDate: 'OrderDate';
            export declare const OrderShippedDate: 'OrderShippedDate';
            export declare const OrderShipVia: 'OrderShipVia';
            export declare const OrderShipCity: 'OrderShipCity';
            export declare const OrderShipCountry: 'OrderShipCountry';
            export declare const ProductName: 'ProductName';
            export declare const ProductDiscontinued: 'ProductDiscontinued';
            export declare const ProductSupplierID: 'ProductSupplierID';
            export declare const ProductQuantityPerUnit: 'ProductQuantityPerUnit';
            export declare const ProductUnitPrice: 'ProductUnitPrice';
            export declare const LineTotal: 'LineTotal';
        }

        ['DetailID', 'OrderID', 'ProductID', 'UnitPrice', 'Quantity', 'Discount', 'OrderCustomerID', 'OrderEmployeeID', 'OrderDate', 'OrderShippedDate', 'OrderShipVia', 'OrderShipCity', 'OrderShipCountry', 'ProductName', 'ProductDiscontinued', 'ProductSupplierID', 'ProductQuantityPerUnit', 'ProductUnitPrice', 'LineTotal'].forEach(x => (<any>Fields)[x] = x);
    }
}

